pkg/spoke/managedcluster/creating_controller.go

```diff
@@ -53,40 +53,80 @@ func NewManagedClusterCreatingController(
 }
 
 func (c *managedClusterCreatingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	_, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
-	switch {
-	case errors.IsUnauthorized(err),
-		errors.IsForbidden(err) && strings.Contains(err.Error(), anonymous):
+	existingCluster, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
+	if err != nil && (errors.IsUnauthorized(err) || errors.IsForbidden(err) && strings.Contains(err.Error(), anonymous)) {
 		klog.V(4).Infof("unable to get the managed cluster %q from hub: %v", c.clusterName, err)
 		return nil
-	case errors.IsNotFound(err):
-	case err == nil:
-		return nil
-	case err != nil:
+	}
+
+	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
 
-	managedCluster := &clusterv1.ManagedCluster{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: c.clusterName,
-		},
+	// create ManagedCluster if not found
+	if err != nil && errors.IsNotFound(err) {
+		managedCluster := &clusterv1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: c.clusterName,
+			},
+		}
+
+		if len(c.spokeExternalServerURLs) != 0 {
+			var managedClusterClientConfigs []clusterv1.ClientConfig
+			for _, serverURL := range c.spokeExternalServerURLs {
+				managedClusterClientConfigs = append(managedClusterClientConfigs, clusterv1.ClientConfig{
+					URL:      serverURL,
+					CABundle: c.spokeCABundle,
+				})
+			}
+			managedCluster.Spec.ManagedClusterClientConfigs = managedClusterClientConfigs
+		}
+
+		_, err = c.hubClusterClient.ClusterV1().ManagedClusters().Create(ctx, managedCluster, metav1.CreateOptions{})
+		if err != nil {
```

Comment:
I think you would want to skip unauthorized here also?

It is better if we have a `func skipUnauthorizedError(err error) error`. So we can use it when run get/update/create

Reply:
ok

Reply:
@qiujian16 done!